File: samples/client/wordnik/scala/src/main/scala/io/swagger/client/api/WordsApi.scala
package io.swagger.client.api

import scala.collection.mutable.HashMap

class WordsApi(val defBasePath: String = "https://api.wordnik.com/v4",
               defApiInvoker: ApiInvoker = ApiInvoker) {
  var basePath = defBasePath
  var apiInvoker = defApiInvoker

  def addHeader(key: String, value: String) = apiInvoker.defaultHeaders += key -> value


  def getRandomWord(hasDictionaryDef: String, includePartOfSpeech: String, excludePartOfSpeech: String, minCorpusCount: Integer, maxCorpusCount: Integer, minDictionaryCount: Integer, maxDictionaryCount: Integer, minLength: Integer, maxLength: Integer): Option[WordObject] = {
    // create path and map variables
    val path = "/words.json/randomWord".replaceAll("\\{format\\}", "json")


    val contentType = {

      "application/json"
    }

    // query params
    val queryParams = new HashMap[String, String]
    val headerParams = new HashMap[String, String]



    if (String.valueOf(hasDictionaryDef) != "null") queryParams += "hasDictionaryDef" -> hasDictionaryDef.toString
    if (String.valueOf(includePartOfSpeech) != "null") queryParams += "includePartOfSpeech" -> includePartOfSpeech.toString
    if (String.valueOf(excludePartOfSpeech) != "null") queryParams += "excludePartOfSpeech" -> excludePartOfSpeech.toString
    if (String.valueOf(minCorpusCount) != "null") queryParams += "minCorpusCount" -> minCorpusCount.toString
    if (String.valueOf(maxCorpusCount) != "null") queryParams += "maxCorpusCount" -> maxCorpusCount.toString
    if (String.valueOf(minDictionaryCount) != "null") queryParams += "minDictionaryCount" -> minDictionaryCount.toString
    if (String.valueOf(maxDictionaryCount) != "null") queryParams += "maxDictionaryCount" -> maxDictionaryCount.toString
    if (String.valueOf(minLength) != "null") queryParams += "minLength" -> minLength.toString
    if (String.valueOf(maxLength) != "null") queryParams += "maxLength" -> maxLength.toString




    try {
      apiInvoker.invokeApi(basePath, path, "GET", queryParams.toMap, None, headerParams.toMap, contentType) match {
        case s: String =>
          Some(ApiInvoker.deserialize(s, "", classOf[WordObject]).asInstanceOf[WordObject])

        case _ => None
      }
    } catch {
      case ex: ApiException if ex.code == 404 => None
      case ex: ApiException => throw ex
    }
  }

  def getRandomWords(hasDictionaryDef: String, includePartOfSpeech: String, excludePartOfSpeech: String, minCorpusCount: Integer, maxCorpusCount: Integer, minDictionaryCount: Integer, maxDictionaryCount: Integer, minLength: Integer, maxLength: Integer, sortBy: String, sortOrder: String, limit: Integer) = {
    // create path and map variables
    val path = "/words.json/randomWords".replaceAll("\\{format\\}", "json")


    val contentType = {

      "application/json"
    }

    // query params
    val queryParams = new HashMap[String, String]
    val headerParams = new HashMap[String, String]



    if (String.valueOf(hasDictionaryDef) != "null") queryParams += "hasDictionaryDef" -> hasDictionaryDef.toString
    if (String.valueOf(includePartOfSpeech) != "null") queryParams += "includePartOfSpeech" -> includePartOfSpeech.toString
    if (String.valueOf(excludePartOfSpeech) != "null") queryParams += "excludePartOfSpeech" -> excludePartOfSpeech.toString
    if (String.valueOf(minCorpusCount) != "null") queryParams += "minCorpusCount" -> minCorpusCount.toString
    if (String.valueOf(maxCorpusCount) != "null") queryParams += "maxCorpusCount" -> maxCorpusCount.toString
    if (String.valueOf(minDictionaryCount) != "null") queryParams += "minDictionaryCount" -> minDictionaryCount.toString
    if (String.valueOf(maxDictionaryCount) != "null") queryParams += "maxDictionaryCount" -> maxDictionaryCount.toString
    if (String.valueOf(minLength) != "null") queryParams += "minLength" -> minLength.toString
    if (String.valueOf(maxLength) != "null") queryParams += "maxLength" -> maxLength.toString
    if (String.valueOf(sortBy) != "null") queryParams += "sortBy" -> sortBy.toString
    if (String.valueOf(sortOrder) != "null") queryParams += "sortOrder" -> sortOrder.toString
    if (String.valueOf(limit) != "null") queryParams += "limit" -> limit.toString




    try {
      apiInvoker.invokeApi(basePath, path, "GET", queryParams.toMap, None, headerParams.toMap, contentType) match {
        case s: String =>

        case _ => None
      }
    } catch {
      case ex: ApiException if ex.code == 404 => None
      case ex: ApiException => throw ex
    }
  }

  def reverseDictionary(query: String, findSenseForWord: String, includeSourceDictionaries: String, excludeSourceDictionaries: String, includePartOfSpeech: String, excludePartOfSpeech: String, minCorpusCount: Integer, maxCorpusCount: Integer, minLength: Integer, maxLength: Integer, expandTerms: String, includeTags: String, sortBy: String, sortOrder: String, skip: String, limit: Integer): Option[DefinitionSearchResults] = {
    // create path and map variables
    val path = "/words.json/reverseDictionary".replaceAll("\\{format\\}", "json")


    val contentType = {

      "application/json"
    }

    // query params
    val queryParams = new HashMap[String, String]
    val headerParams = new HashMap[String, String]



    if (String.valueOf(query) != "null") queryParams += "query" -> query.toString
    if (String.valueOf(findSenseForWord) != "null") queryParams += "findSenseForWord" -> findSenseForWord.toString
    if (String.valueOf(includeSourceDictionaries) != "null") queryParams += "includeSourceDictionaries" -> includeSourceDictionaries.toString
    if (String.valueOf(excludeSourceDictionaries) != "null") queryParams += "excludeSourceDictionaries" -> excludeSourceDictionaries.toString
    if (String.valueOf(includePartOfSpeech) != "null") queryParams += "includePartOfSpeech" -> includePartOfSpeech.toString
    if (String.valueOf(excludePartOfSpeech) != "null") queryParams += "excludePartOfSpeech" -> excludePartOfSpeech.toString
    if (String.valueOf(minCorpusCount) != "null") queryParams += "minCorpusCount" -> minCorpusCount.toString
    if (String.valueOf(maxCorpusCount) != "null") queryParams += "maxCorpusCount" -> maxCorpusCount.toString
    if (String.valueOf(minLength) != "null") queryParams += "minLength" -> minLength.toString
    if (String.valueOf(maxLength) != "null") queryParams += "maxLength" -> maxLength.toString
    if (String.valueOf(expandTerms) != "null") queryParams += "expandTerms" -> expandTerms.toString
    if (String.valueOf(includeTags) != "null") queryParams += "includeTags" -> includeTags.toString
    if (String.valueOf(sortBy) != "null") queryParams += "sortBy" -> sortBy.toString
    if (String.valueOf(sortOrder) != "null") queryParams += "sortOrder" -> sortOrder.toString
    if (String.valueOf(skip) != "null") queryParams += "skip" -> skip.toString
    if (String.valueOf(limit) != "null") queryParams += "limit" -> limit.toString




    try {
      apiInvoker.invokeApi(basePath, path, "GET", queryParams.toMap, None, headerParams.toMap, contentType) match {
        case s: String =>
          Some(ApiInvoker.deserialize(s, "", classOf[DefinitionSearchResults]).asInstanceOf[DefinitionSearchResults])

        case _ => None
      }
    } catch {
      case ex: ApiException if ex.code == 404 => None
      case ex: ApiException => throw ex
    }
  }

  def searchWords(query: String, caseSensitive: String, includePartOfSpeech: String, excludePartOfSpeech: String, minCorpusCount: Integer, maxCorpusCount: Integer, minDictionaryCount: Integer, maxDictionaryCount: Integer, minLength: Integer, maxLength: Integer, skip: Integer, limit: Integer): Option[WordSearchResults] = {
    // create path and map variables
    val path = "/words.json/search/{query}".replaceAll("\\{format\\}", "json").replaceAll("\\{" + "query" + "\\}", apiInvoker.escape(query))




    val contentType = {

      "application/json"
    }

    // query params
    val queryParams = new HashMap[String, String]
    val headerParams = new HashMap[String, String]



    if (String.valueOf(caseSensitive) != "null") queryParams += "caseSensitive" -> caseSensitive.toString
    if (String.valueOf(includePartOfSpeech) != "null") queryParams += "includePartOfSpeech" -> includePartOfSpeech.toString
    if (String.valueOf(excludePartOfSpeech) != "null") queryParams += "excludePartOfSpeech" -> excludePartOfSpeech.toString
    if (String.valueOf(minCorpusCount) != "null") queryParams += "minCorpusCount" -> minCorpusCount.toString
    if (String.valueOf(maxCorpusCount) != "null") queryParams += "maxCorpusCount" -> maxCorpusCount.toString
    if (String.valueOf(minDictionaryCount) != "null") queryParams += "minDictionaryCount" -> minDictionaryCount.toString
    if (String.valueOf(maxDictionaryCount) != "null") queryParams += "maxDictionaryCount" -> maxDictionaryCount.toString
    if (String.valueOf(minLength) != "null") queryParams += "minLength" -> minLength.toString
    if (String.valueOf(maxLength) != "null") queryParams += "maxLength" -> maxLength.toString
    if (String.valueOf(skip) != "null") queryParams += "skip" -> skip.toString
    if (String.valueOf(limit) != "null") queryParams += "limit" -> limit.toString




    try {
      apiInvoker.invokeApi(basePath, path, "GET", queryParams.toMap, None, headerParams.toMap, contentType) match {
        case s: String =>
          Some(ApiInvoker.deserialize(s, "", classOf[WordSearchResults]).asInstanceOf[WordSearchResults])

        case _ => None
      }
    } catch {
      case ex: ApiException if ex.code == 404 => None
      case ex: ApiException => throw ex
    }
  }

  def getWordOfTheDay(date: String): Option[WordOfTheDay] = {
    // create path and map variables
    val path = "/words.json/wordOfTheDay".replaceAll("\\{format\\}", "json")


    val contentType = {

      "application/json"
    }

    // query params
    val queryParams = new HashMap[String, String]
    val headerParams = new HashMap[String, String]



    if (String.valueOf(date) != "null") queryParams += "date" -> date.toString




    try {
      apiInvoker.invokeApi(basePath, path, "GET", queryParams.toMap, None, headerParams.toMap, contentType) match {
        case s: String =>
          Some(ApiInvoker.deserialize(s, "", classOf[WordOfTheDay]).asInstanceOf[WordOfTheDay])

        case _ => None
      }
    } catch {
      case ex: ApiException if ex.code == 404 => None
      case ex: ApiException => throw ex
    }
  }

}

File: samples/client/wordnik/java/src/main/java/io/swagger/client/api/WordsApi.java
package io.swagger.client.api;

import com.sun.jersey.multipart.FormDataMultiPart;
import io.swagger.client.ApiException;
import io.swagger.client.ApiInvoker;
import io.swagger.client.model.DefinitionSearchResults;
import io.swagger.client.model.WordObject;
import io.swagger.client.model.WordOfTheDay;
import io.swagger.client.model.WordSearchResults;

import java.util.HashMap;
import java.util.Map;

public class WordsApi {
    String basePath = "https://api.wordnik.com/v4";
    ApiInvoker apiInvoker = ApiInvoker.getInstance();

    public ApiInvoker getInvoker() {
        return apiInvoker;
    }

    public String getBasePath() {
        return basePath;
    }

    public void setBasePath(String basePath) {
        this.basePath = basePath;
    }

    public WordObject getRandomWord(String hasDictionaryDef, String includePartOfSpeech, String excludePartOfSpeech, Integer minCorpusCount, Integer maxCorpusCount, Integer minDictionaryCount, Integer maxDictionaryCount, Integer minLength, Integer maxLength) throws ApiException {
        Object postBody = null;


        // create path and map variables
        String path = "/words.json/randomWord".replaceAll("\\{format\\}", "json");

        // query params
        Map<String, String> queryParams = new HashMap<String, String>();
        Map<String, String> headerParams = new HashMap<String, String>();
        Map<String, String> formParams = new HashMap<String, String>();

        if (!"null".equals(String.valueOf(hasDictionaryDef))) {
            queryParams.put("hasDictionaryDef", String.valueOf(hasDictionaryDef));
        }
        if (!"null".equals(String.valueOf(includePartOfSpeech))) {
            queryParams.put("includePartOfSpeech", String.valueOf(includePartOfSpeech));
        }
        if (!"null".equals(String.valueOf(excludePartOfSpeech))) {
            queryParams.put("excludePartOfSpeech", String.valueOf(excludePartOfSpeech));
        }
        if (!"null".equals(String.valueOf(minCorpusCount))) {
            queryParams.put("minCorpusCount", String.valueOf(minCorpusCount));
        }
        if (!"null".equals(String.valueOf(maxCorpusCount))) {
            queryParams.put("maxCorpusCount", String.valueOf(maxCorpusCount));
        }
        if (!"null".equals(String.valueOf(minDictionaryCount))) {
            queryParams.put("minDictionaryCount", String.valueOf(minDictionaryCount));
        }
        if (!"null".equals(String.valueOf(maxDictionaryCount))) {
            queryParams.put("maxDictionaryCount", String.valueOf(maxDictionaryCount));
        }
        if (!"null".equals(String.valueOf(minLength))) {
            queryParams.put("minLength", String.valueOf(minLength));
        }
        if (!"null".equals(String.valueOf(maxLength))) {
            queryParams.put("maxLength", String.valueOf(maxLength));
        }


        String[] contentTypes = {

        };

        String contentType = contentTypes.length > 0 ? contentTypes[0] : "application/json";

        if (contentType.startsWith("multipart/form-data")) {
            boolean hasFields = false;
            FormDataMultiPart mp = new FormDataMultiPart();

            if (hasFields) {
                postBody = mp;
            }
        } else {

        }

        try {
            String response = apiInvoker.invokeAPI(basePath, path, "GET", queryParams, postBody, headerParams, formParams, contentType);
            if (response != null) {
                return (WordObject) ApiInvoker.deserialize(response, "", WordObject.class);
            } else {
                return null;
            }
        } catch (ApiException ex) {
            if (ex.getCode() == 404) {
                return null;
            } else {
                throw ex;
            }
        }
    }


    public void getRandomWords(String hasDictionaryDef, String includePartOfSpeech, String excludePartOfSpeech, Integer minCorpusCount, Integer maxCorpusCount, Integer minDictionaryCount, Integer maxDictionaryCount, Integer minLength, Integer maxLength, String sortBy, String sortOrder, Integer limit) throws ApiException {
        Object postBody = null;


        // create path and map variables
        String path = "/words.json/randomWords".replaceAll("\\{format\\}", "json");

        // query params
        Map<String, String> queryParams = new HashMap<String, String>();
        Map<String, String> headerParams = new HashMap<String, String>();
        Map<String, String> formParams = new HashMap<String, String>();

        if (!"null".equals(String.valueOf(hasDictionaryDef))) {
            queryParams.put("hasDictionaryDef", String.valueOf(hasDictionaryDef));
        }
        if (!"null".equals(String.valueOf(includePartOfSpeech))) {
            queryParams.put("includePartOfSpeech", String.valueOf(includePartOfSpeech));
        }
        if (!"null".equals(String.valueOf(excludePartOfSpeech))) {
            queryParams.put("excludePartOfSpeech", String.valueOf(excludePartOfSpeech));
        }
        if (!"null".equals(String.valueOf(minCorpusCount))) {
            queryParams.put("minCorpusCount", String.valueOf(minCorpusCount));
        }
        if (!"null".equals(String.valueOf(maxCorpusCount))) {
            queryParams.put("maxCorpusCount", String.valueOf(maxCorpusCount));
        }
        if (!"null".equals(String.valueOf(minDictionaryCount))) {
            queryParams.put("minDictionaryCount", String.valueOf(minDictionaryCount));
        }
        if (!"null".equals(String.valueOf(maxDictionaryCount))) {
            queryParams.put("maxDictionaryCount", String.valueOf(maxDictionaryCount));
        }
        if (!"null".equals(String.valueOf(minLength))) {
            queryParams.put("minLength", String.valueOf(minLength));
        }
        if (!"null".equals(String.valueOf(maxLength))) {
            queryParams.put("maxLength", String.valueOf(maxLength));
        }
        if (!"null".equals(String.valueOf(sortBy))) {
            queryParams.put("sortBy", String.valueOf(sortBy));
        }
        if (!"null".equals(String.valueOf(sortOrder))) {
            queryParams.put("sortOrder", String.valueOf(sortOrder));
        }
        if (!"null".equals(String.valueOf(limit))) {
            queryParams.put("limit", String.valueOf(limit));
        }


        String[] contentTypes = {

        };

        String contentType = contentTypes.length > 0 ? contentTypes[0] : "application/json";

        if (contentType.startsWith("multipart/form-data")) {
            boolean hasFields = false;
            FormDataMultiPart mp = new FormDataMultiPart();

            if (hasFields) {
                postBody = mp;
            }
        } else {

        }

        try {
            String response = apiInvoker.invokeAPI(basePath, path, "GET", queryParams, postBody, headerParams, formParams, contentType);
            if (response != null) {
                return;
            } else {
                return;
            }
        } catch (ApiException ex) {
            if (ex.getCode() == 404) {
                return;
            } else {
                throw ex;
            }
        }
    }


    public DefinitionSearchResults reverseDictionary(String query, String findSenseForWord, String includeSourceDictionaries, String excludeSourceDictionaries, String includePartOfSpeech, String excludePartOfSpeech, Integer minCorpusCount, Integer maxCorpusCount, Integer minLength, Integer maxLength, String expandTerms, String includeTags, String sortBy, String sortOrder, String skip, Integer limit) throws ApiException {
        Object postBody = null;


        // create path and map variables
        String path = "/words.json/reverseDictionary".replaceAll("\\{format\\}", "json");

        // query params
        Map<String, String> queryParams = new HashMap<String, String>();
        Map<String, String> headerParams = new HashMap<String, String>();
        Map<String, String> formParams = new HashMap<String, String>();

        if (!"null".equals(String.valueOf(query))) {
            queryParams.put("query", String.valueOf(query));
        }
        if (!"null".equals(String.valueOf(findSenseForWord))) {
            queryParams.put("findSenseForWord", String.valueOf(findSenseForWord));
        }
        if (!"null".equals(String.valueOf(includeSourceDictionaries))) {
            queryParams.put("includeSourceDictionaries", String.valueOf(includeSourceDictionaries));
        }
        if (!"null".equals(String.valueOf(excludeSourceDictionaries))) {
            queryParams.put("excludeSourceDictionaries", String.valueOf(excludeSourceDictionaries));
        }
        if (!"null".equals(String.valueOf(includePartOfSpeech))) {
            queryParams.put("includePartOfSpeech", String.valueOf(includePartOfSpeech));
        }
        if (!"null".equals(String.valueOf(excludePartOfSpeech))) {
            queryParams.put("excludePartOfSpeech", String.valueOf(excludePartOfSpeech));
        }
        if (!"null".equals(String.valueOf(minCorpusCount))) {
            queryParams.put("minCorpusCount", String.valueOf(minCorpusCount));
        }
        if (!"null".equals(String.valueOf(maxCorpusCount))) {
            queryParams.put("maxCorpusCount", String.valueOf(maxCorpusCount));
        }
        if (!"null".equals(String.valueOf(minLength))) {
            queryParams.put("minLength", String.valueOf(minLength));
        }
        if (!"null".equals(String.valueOf(maxLength))) {
            queryParams.put("maxLength", String.valueOf(maxLength));
        }
        if (!"null".equals(String.valueOf(expandTerms))) {
            queryParams.put("expandTerms", String.valueOf(expandTerms));
        }
        if (!"null".equals(String.valueOf(includeTags))) {
            queryParams.put("includeTags", String.valueOf(includeTags));
        }
        if (!"null".equals(String.valueOf(sortBy))) {
            queryParams.put("sortBy", String.valueOf(sortBy));
        }
        if (!"null".equals(String.valueOf(sortOrder))) {
            queryParams.put("sortOrder", String.valueOf(sortOrder));
        }
        if (!"null".equals(String.valueOf(skip))) {
            queryParams.put("skip", String.valueOf(skip));
        }
        if (!"null".equals(String.valueOf(limit))) {
            queryParams.put("limit", String.valueOf(limit));
        }


        String[] contentTypes = {

        };

        String contentType = contentTypes.length > 0 ? contentTypes[0] : "application/json";

        if (contentType.startsWith("multipart/form-data")) {
            boolean hasFields = false;
            FormDataMultiPart mp = new FormDataMultiPart();

            if (hasFields) {
                postBody = mp;
            }
        } else {

        }

        try {
            String response = apiInvoker.invokeAPI(basePath, path, "GET", queryParams, postBody, headerParams, formParams, contentType);
            if (response != null) {
                return (DefinitionSearchResults) ApiInvoker.deserialize(response, "", DefinitionSearchResults.class);
            } else {
                return null;
            }
        } catch (ApiException ex) {
            if (ex.getCode() == 404) {
                return null;
            } else {
                throw ex;
            }
        }
    }


    public WordSearchResults searchWords(String query, String caseSensitive, String includePartOfSpeech, String excludePartOfSpeech, Integer minCorpusCount, Integer maxCorpusCount, Integer minDictionaryCount, Integer maxDictionaryCount, Integer minLength, Integer maxLength, Integer skip, Integer limit) throws ApiException {
        Object postBody = null;


        // create path and map variables
        String path = "/words.json/search/{query}".replaceAll("\\{format\\}", "json")
                .replaceAll("\\{" + "query" + "\\}", apiInvoker.escapeString(query.toString()));

        // query params
        Map<String, String> queryParams = new HashMap<String, String>();
        Map<String, String> headerParams = new HashMap<String, String>();
        Map<String, String> formParams = new HashMap<String, String>();

        if (!"null".equals(String.valueOf(caseSensitive))) {
            queryParams.put("caseSensitive", String.valueOf(caseSensitive));
        }
        if (!"null".equals(String.valueOf(includePartOfSpeech))) {
            queryParams.put("includePartOfSpeech", String.valueOf(includePartOfSpeech));
        }
        if (!"null".equals(String.valueOf(excludePartOfSpeech))) {
            queryParams.put("excludePartOfSpeech", String.valueOf(excludePartOfSpeech));
        }
        if (!"null".equals(String.valueOf(minCorpusCount))) {
            queryParams.put("minCorpusCount", String.valueOf(minCorpusCount));
        }
        if (!"null".equals(String.valueOf(maxCorpusCount))) {
            queryParams.put("maxCorpusCount", String.valueOf(maxCorpusCount));
        }
        if (!"null".equals(String.valueOf(minDictionaryCount))) {
            queryParams.put("minDictionaryCount", String.valueOf(minDictionaryCount));
        }
        if (!"null".equals(String.valueOf(maxDictionaryCount))) {
            queryParams.put("maxDictionaryCount", String.valueOf(maxDictionaryCount));
        }
        if (!"null".equals(String.valueOf(minLength))) {
            queryParams.put("minLength", String.valueOf(minLength));
        }
        if (!"null".equals(String.valueOf(maxLength))) {
            queryParams.put("maxLength", String.valueOf(maxLength));
        }
        if (!"null".equals(String.valueOf(skip))) {
            queryParams.put("skip", String.valueOf(skip));
        }
        if (!"null".equals(String.valueOf(limit))) {
            queryParams.put("limit", String.valueOf(limit));
        }


        String[] contentTypes = {

        };

        String contentType = contentTypes.length > 0 ? contentTypes[0] : "application/json";

        if (contentType.startsWith("multipart/form-data")) {
            boolean hasFields = false;
            FormDataMultiPart mp = new FormDataMultiPart();

            if (hasFields) {
                postBody = mp;
            }
        } else {

        }

        try {
            String response = apiInvoker.invokeAPI(basePath, path, "GET", queryParams, postBody, headerParams, formParams, contentType);
            if (response != null) {
                return (WordSearchResults) ApiInvoker.deserialize(response, "", WordSearchResults.class);
            } else {
                return null;
            }
        } catch (ApiException ex) {
            if (ex.getCode() == 404) {
                return null;
            } else {
                throw ex;
            }
        }
    }


    public WordOfTheDay getWordOfTheDay(String date) throws ApiException {
        Object postBody = null;


        // create path and map variables
        String path = "/words.json/wordOfTheDay".replaceAll("\\{format\\}", "json");

        // query params
        Map<String, String> queryParams = new HashMap<String, String>();
        Map<String, String> headerParams = new HashMap<String, String>();
        Map<String, String> formParams = new HashMap<String, String>();

        if (!"null".equals(String.valueOf(date))) {
            queryParams.put("date", String.valueOf(date));
        }


        String[] contentTypes = {

        };

        String contentType = contentTypes.length > 0 ? contentTypes[0] : "application/json";

        if (contentType.startsWith("multipart/form-data")) {
            boolean hasFields = false;
            FormDataMultiPart mp = new FormDataMultiPart();

            if (hasFields) {
                postBody = mp;
            }
        } else {

        }

        try {
            String response = apiInvoker.invokeAPI(basePath, path, "GET", queryParams, postBody, headerParams, formParams, contentType);
            if (response != null) {
                return (WordOfTheDay) ApiInvoker.deserialize(response, "", WordOfTheDay.class);
            } else {
                return null;
            }
        } catch (ApiException ex) {
            if (ex.getCode() == 404) {
                return null;
            } else {
                throw ex;
            }
        }
    }

}
